### DISCUSSION TITLE
BasicPlanner doesn't work with 'from semantic_kernel.connectors.ai.open_ai import AzureChatCompletion'

### DISCUSSION THREAD

**Discussion by yuichiromukaiyama** [microsoft/semantic-kernel#5011] [General]
This issue is related to https://github.com/microsoft/semantic-kernel/issues/4923. I am posting this as a question as well, as I have been struggling with how to handle this issue. Here is a minimal code that reproduces the problem.

```python
kernel = sk.Kernel()

kernel.add_chat_service(
    "chat_completion",
    AzureChatCompletion(
        deployment_name=AZURE_OPENAI_DEPLOYMENT_NAME,
        endpoint=AZURE_OPENAI_ENDPOINT,
        api_key=AZURE_OPENAI_API_KEY,
    ),
)

planner = BasicPlanner()
plan = planner.create_plan(goal=goal, kernel=kernel)
```

When I run it, I get the error `'Choice' object has no attribute 'logprobs'` as in Issue.
Investigating the cause, I found that BasicPlanner is calling `create_semantic_function` internally.

https://github.com/microsoft/semantic-kernel/blob/1490eb6d02f874ffabe9549d917421910886e355/python/semantic_kernel/planning/basic_planner.py#L186

Further tracing inside reveals the following line, which uses `PromptTemplate`.

https://github.com/microsoft/semantic-kernel/blob/1490eb6d02f874ffabe9549d917421910886e355/python/semantic_kernel/kernel.py#L870

This causes harm in the following places `has_chat_prompt` is always `False` for `PromptTemplate`.

https://github.com/microsoft/semantic-kernel/blob/1490eb6d02f874ffabe9549d917421910886e355/python/semantic_kernel/kernel.py#L727

```python
@dataclass
class SemanticFunctionConfig:
    prompt_template_config: "PromptTemplateConfig"
    prompt_template: "PromptTemplate"

    @property
    def has_chat_prompt(self) -> bool:
        return isinstance(self.prompt_template, ChatPromptTemplate)
```

Therefore, `AzureChatCompletion` is not used when creating a plan in BasicPlanner. As proof, I change `PromptTemplate` to `ChatPromptTemplate` and run it and it works fine.

Is there a workable solution to this? I am thinking as follows

1. https://github.com/microsoft/semantic-kernel/issues/4923 made 'logprobs' nullable for ChatCompletion only. Publish a PR that modifies TextCompletion as well.

2. Publish a PR to change 'PromptTemplate' to 'ChatPromptTemplate' according to chat-service.

3. BasicPlanner uses the principle TextCompletion or AzureTextCompletion

※ In case 3, I think it's better to change the Quick Start.
https://github.com/microsoft/semantic-kernel/blob/main/python/notebooks/05-using-the-planner.ipynb

**Comment by yuichiromukaiyama** [comment]
same problem

https://github.com/microsoft/semantic-kernel/pull/4924#issuecomment-1951796494

**Comment by ymuichiro** [comment]
PR

https://github.com/microsoft/semantic-kernel/pull/5071/files

**Reply by ymuichiro** [reply]
Merged

**Comment by moonbox3** [comment]
Hi @yuichiromukaiyama, please, in the future, open this as a GitHub issue instead of a Discussion. We'll be able to get to this much faster. Thank you!

**Reply by yuichiromukaiyama** [reply]
From now on, I will write directly to the issue. Thank you!
